src/tools/sfdk/sdkmanager.cpp

```diff
@@ -1348,10 +1348,22 @@ bool SdkManager::mapEnginePaths(QString *program, QStringList *arguments, QStrin
         {cleanSharedSrc, Constants::BUILD_ENGINE_SHARED_SRC_MOUNT_POINT, caseInsensitiveOnWindows}
 #endif
     };
+    *program = QDir::fromNativeSeparators(*program);
+
     for (const Mapping &mapping : mappings) {
         qCDebug(sfdk) << "Mapping" << mapping.hostPath << "as" << mapping.enginePath;
         program->replace(mapping.hostPath, mapping.enginePath, mapping.cs);
-        arguments->replaceInStrings(mapping.hostPath, mapping.enginePath, mapping.cs);
+        QRegularExpression::PatternOption reCaseSensitivity = (mapping.cs == Qt::CaseInsensitive)
+                ? QRegularExpression::CaseInsensitiveOption : QRegularExpression::NoPatternOption;
+        QRegularExpression re(QRegularExpression::escape(mapping.hostPath) + "([^:;,'\"]*)",
+                              reCaseSensitivity);
+        arguments->replaceInStrings(re, mapping.enginePath + "\\1");
+        QString nativeHostPath = QDir::toNativeSeparators(mapping.hostPath);
+        if (mapping.hostPath != nativeHostPath) {
+            QRegularExpression nativeRe(QRegularExpression::escape(nativeHostPath) + "([^:;,'\"]*)",
+                                        reCaseSensitivity);
+            arguments->replaceInStrings(nativeRe, QDir::fromNativeSeparators(mapping.enginePath + "\\1"));
+        }
```

Comment:
(Comment written before your last update)

We somehow forgot about replacing the backslashes in the captured group.

The most naive approach would be

```
auto map = [=](QString string) {
    while (string.contains(re, &match)) {
        QString replacement = mapping.enginePath + QDir::fromNativeSeparators(match.captured(1));
        return string.replace(match.captured(), replacement);
    }
};
*arguments = Utils::transform(*arguments, map);
```

And we need to also map in environment variables

```
environment->insert(key, map(environment->value(key)));
```

Reply:
For the alternative RE I would consider combining that with

```
QRegularExpression re(QString("(?:%1|%2)([^:;,'\"]*)")
        .arg(QRegularExpression::escape(mapping.hostPath))
        .arg(QRegularExpression::escape(QDir::toNativeSeparators(mapping.hostPath))),
        reCaseSensitivity);
````